python/docs/dataset.rst

```diff
@@ -0,0 +1,8 @@
+Lance Dataset
```

Comment:
So we will migrate to rtd first and then migrating to mardown, right?

Reply:
this will just be in rtd, it's just easier navigation for the generated doc so I kept it there. In general in the main Lance doc my expectation is that we put multi-language blocks for all our code examples